File: DSA (Add Code Here)/linked_list.c
#include <stdio.h>
#include <stdlib.h>

struct Node{   //Creating a node // node comprises head and next(address of next head)
    int data;
    struct Node *next;
}*head;

// Inserting element at the beginning
void insert_begin(int x){
    struct Node *new_node;
    new_node = malloc(sizeof(struct Node*));
    new_node -> data = x;
    if(head == NULL){
        head = new_node;
        new_node -> next = NULL;
    }
    else{
        new_node -> next = head;
        head = new_node;
    }
}

void print_list(struct Node *n){   // to display the contents of linked list
    while (n != NULL){
        printf("%d  ",n->data);
        n = n->next;
    }
}

int main(){
    //struct Node *head = NULL;  // Since we're inserting at front, hence cannot assign HEAD as NULL.
    struct Node *second = NULL;
    struct Node *third = NULL;

    head = malloc(sizeof(struct Node));
    second = malloc(sizeof(struct Node));
    third = malloc(sizeof(struct Node));


    head -> data = 10;
    head -> next = second;
    second -> data = 20;
    second -> next = third;
    third -> data = 30;
    third -> next = NULL;

    insert_begin(50);

    //print_list(head);

    //insert_begin(5);

    print_list(head);

    return 0;
}

File: Hacktoberfest2023-main/Add Code Here/linkedlist.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class linkedlist:
    def __init__(self):
        self.head = None

    def append(self, data):
        new_node = Node(data)
        if not self.head:
            self.head = new_node
            return
        current = self.head
        while current.next:
            current = current.next
        current.next = new_node

    def display(self):
        current = self.head
        while current:
            print(current.data, end=" -> ")
            current = current.next
        print("None")


my_linked_list = LinkedList()

my_linked_list.append(1)
my_linked_list.append(2)
my_linked_list.append(3)
my_linked_list.append(4)

print("Linked List:")
my_linked_list.display()
